File: frontend/src/lib/api/ai.ts
import { apiClient } from "@/lib/api/client";
import type { AIProvidersResponse, AIRequest, AIResponse } from "@/lib/api/types";

// Maps to backend/app/api/v1/ai.py -> POST /ai/complete
export async function completeWithAI(payload: AIRequest): Promise<AIResponse> {
  const { data } = await apiClient.post<AIResponse>("/ai/complete", payload);
  return data;
}

// GET /ai/providers
export async function listAIProviders(): Promise<AIProvidersResponse> {
  const { data } = await apiClient.get<AIProvidersResponse>("/ai/providers");
  return data;
}

File: backend/app/api/v1/ai.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.v1.schemas import AIRequest, AIResponse
from app.database.database import get_session
from app.services.ai_service import AIService

router = APIRouter()


@router.post("/complete", response_model=AIResponse)
async def complete(payload: AIRequest, session: AsyncSession = Depends(get_session)):
    return await AIService(session).answer(
        company_id=payload.company_id,
        prompt=payload.prompt,
        provider=payload.provider,
        model=payload.model,
    )

@router.get("/providers")
async def providers():
    return {
        "default": "mock",
        "available": ["mock", "openai", "claude", "gemini", "grok", "deepseek", "ollama"],
    }

